contracts/modules/NftTransfer.sol

```diff
@@ -0,0 +1,136 @@
+pragma solidity ^0.5.4;
+
+import "./common/BaseModule.sol";
+import "./common/RelayerModule.sol";
+import "../storage/GuardianStorage.sol";
+
+/**
+ * @title NftTransfer
+ * @dev Module to transfer NFTs (ERC721),
+ * @author Olivier VDB - <olivier@argent.xyz>
+ */
+contract NftTransfer is BaseModule, RelayerModule {
+
+    bytes32 constant NAME = "NftTransfer";
+
+    // Equals to `bytes4(keccak256("onERC721Received(address,address,uint256,bytes)"))`
+    bytes4 private constant ERC721_RECEIVED = 0x150b7a02;
+
+    // The Guardian storage 
+    GuardianStorage public guardianStorage;
+
+    // *************** Events *************************** //
+
+    event NonFungibleTransfer(address indexed wallet, address indexed nftContract, uint256 indexed tokenId, address to, bytes data);    
+
+    // *************** Modifiers *************************** //
+
+    /**
+     * @dev Throws if the caller is not the owner or an authorised module.
+     */
+    modifier onlyOwnerOrModule(BaseWallet _wallet) {
+        require(isOwner(_wallet, msg.sender) || _wallet.authorised(msg.sender), "NT: must be wallet owner or module");
+        _;
+    }
+
+    /**
+     * @dev Throws if the wallet is locked.
+     */
+    modifier onlyWhenUnlocked(BaseWallet _wallet) {
+        // solium-disable-next-line security/no-block-members
+        require(!guardianStorage.isLocked(_wallet), "NT: wallet must be unlocked");
+        _;
+    }
+
+    // *************** Constructor ********************** //
+
+    constructor(
+        ModuleRegistry _registry,
+        GuardianStorage _guardianStorage
+    ) 
+        BaseModule(_registry, NAME)
+        public 
+    {
+        guardianStorage = _guardianStorage;
+    }
+
+    // *************** External/Public Functions ********************* //
+
+    /**
+     * @dev Inits the module for a wallet by setting up the onERC721Received
+     * static call redirection from the wallet to the module.
+     * @param _wallet The target wallet.
+     */
+    function init(BaseWallet _wallet) external onlyWallet(_wallet) {
+        _wallet.enableStaticCall(address(this), ERC721_RECEIVED);
+    }
+
+    /**
+     * @notice Handle the receipt of an NFT
+     * @dev An ERC721 smart contract calls this function on the recipient contract
+     * after a `safeTransfer`. If the recipient is a BaseWallet, the call to onERC721Received 
+     * will be forwarded to the method onERC721Received of the present module. 
+     * @param operator The address which called `safeTransferFrom` function
+     * @param from The address which previously owned the token
+     * @param tokenId The NFT identifier which is being transferred
+     * @param data Additional data with no specified format
+     * @return bytes4 `bytes4(keccak256("onERC721Received(address,address,uint256,bytes)"))`
+     */
+    function onERC721Received(address operator, address from, uint256 tokenId, bytes calldata data)
```

Comment:
Maybe we could emit a Received event here with (wallet, operator, from, tokenId, data) so as to make it easier for clients to track transfers (i.e. they can just monitor the events of the module instead of monitoring all ERC721 transfer and discarding the non Argent ones).

Reply:
We could but the event would only be emitted for `safeTransfer` so the client would anyways need to monitor ERC721 Transfer events for when unsafe transfers are performed. Maybe we should leave it like that for now and only add this Received event after the client guys confirm that this would indeed make their lives easier?

Reply:
Agreed